src/models/goal/impls.rs

```diff
@@ -0,0 +1,154 @@
+            _ => panic!("Invalid day selection"),
+        }
+    }
+}
+
+impl From<Day> for String {
+    fn from(day: Day) -> Self {
+        info!("From<Days> day: {:?}", day);
+        match day {
+            Day::Fri => "Fri".into(),
+            Day::Sat => "Sat".into(),
+            Day::Sun => "Sun".into(),
+            Day::Mon => "Mon".into(),
+            Day::Tue => "Tue".into(),
+            Day::Wed => "Wed".into(),
+            Day::Thu => "Thu".into(),
+        }
+    }
+}
+
+// Todo 2023-05-05  | Check all these setters - Why are they needed? Why public?
+impl Goal {
+    pub fn new(id: usize) -> Self {
+        Self {
+            id: id.to_string(),
+            title: String::from("Test"),
+            ..Default::default()
+        }
+    }
+
+    pub fn title(mut self, title: &str) -> Self {
+        self.title = title.to_string();
+        self
+    }
+
+    pub fn duration(mut self, min_duration: usize) -> Self {
+        self.min_duration = Some(min_duration);
+        self
+    }
+
+    pub fn repeat(mut self, repetition: Repetition) -> Self {
+        self.repeat = Some(repetition);
+        self
+    }
+
+    pub fn start(mut self, start: NaiveDateTime) -> Self {
+        self.start = Some(start);
+        self
+    }
+
+    pub fn deadline(mut self, deadline: NaiveDateTime) -> Self {
+        self.deadline = Some(deadline);
+        self
+    }
+
+    /// Generates a Task/Increment from a Processed Goal
+    /// **Caution!:*** This can only be done after the Goals have been pre-processed!
+    /// Creates and splits the Goal Timeline into one or more segments, making a Task/Increment for each.
+    /// Depending on the Goal Tag, Task/Increments will also get Tags to help with scheduling order:
+    /// - Optional Tag // Todo! add Regular Tag to simplify?
+    /// - Filler Tag
+    /// - FlexDur Tag
+    /// - FlexNum Tag
+    /// - Budget Tag
+    pub fn generate_tasks(
+        self,
+        calendar_start: NaiveDateTime,
+        calendar_end: NaiveDateTime,
+        counter: &mut usize,
+    ) -> Vec<Task> {
+        let mut tasks: Vec<Task> = Vec::new();
+        if self.tags.contains(&Tag::IgnoreForTaskGeneration) {
+            return tasks;
+        }
+
+        if self.tags.contains(&Tag::Budget) {
+            return tasks;
+        }
+        let start = self.start.unwrap_or(calendar_start);
+        let deadline = self.deadline.unwrap_or(calendar_end);
+
+        let time_slots_iterator = TimeSlotsIterator::new(
+            start,
+            deadline,
+            self.repeat,
+            self.filters.clone(),
+            // Todo! add self.before_time filter
+        );
+        dbg!(&time_slots_iterator);
+
+        for timeline in time_slots_iterator {
+            dbg!(&timeline);
+            let task_id = *counter;
+            *counter += 1;
+            // TODO 2023-05-06  | apply Task::new(...)
+            if !timeline.slots.is_empty() && self.min_duration.is_some() {
```

Comment:
```suggestion
            if timeline.slots.len() > 0 && self.min_duration.is_some() {
```

Reply:
Also Warning happen here by `cargo clippy` :

```bash
warning: length comparison to zero
   --> src/models/goal/impls.rs:120:16
    |
120 |             if timeline.slots.len() > 0 && self.min_duration.is_some() {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^ help: using `!is_empty` is clearer and more explicit: `!timeline.slots.is_empty()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_zero

warning: `zinzen` (lib) generated 2 warnings (run `cargo clippy --fix --lib -p zinzen` to apply 2 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
```

Reply:
Ha yeah I don't _always_ agree with Clippy.

Do as you like:  
- use `!is_empty`
- turn this Clippy warning off.